tests/src/python/test_qgsproviderconnection_base.py

```diff
@@ -184,23 +213,30 @@ def _test_operations(self, md, conn):
                     table = "\"%s\".\"myNewAspatialTable\"" % schema
                 else:
                     table = 'myNewAspatialTable'
-                sql = "INSERT INTO %s (string, long, double, integer, date, datetime, time) VALUES ('QGIS Rocks - \U0001f604', 666, 1.234, 1234, '2019-07-08', '2019-07-08T12:00:12', '12:00:13.00')" % table
+
+                # MSSQL literal syntax for UTF8 requires 'N' prefix
+                sql = "INSERT INTO %s (string_t, long_t, double_t, integer_t, date_t, datetime_t, time_t) VALUES (%s'QGIS Rocks - \U0001f604', 666, 1.234, 1234, '2019-07-08', '2019-07-08T12:00:12', '12:00:13.00')" % (table, 'N' if self.providerKey == 'mssql' else '')
                 res = conn.executeSql(sql)
                 self.assertEqual(res, [])
-                sql = "SELECT string, long, double, integer, date, datetime FROM %s" % table
+                sql = "SELECT string_t, long_t, double_t, integer_t, date_t, datetime_t FROM %s" % table
                 res = conn.executeSql(sql)
                 # GPKG has no type for time and spatialite has no support for dates and time ...
                 if self.providerKey == 'spatialite':
                     self.assertEqual(res, [['QGIS Rocks - \U0001f604', 666, 1.234, 1234, '2019-07-08', '2019-07-08T12:00:12']])
                 else:
                     self.assertEqual(res, [['QGIS Rocks - \U0001f604', 666, 1.234, 1234, QtCore.QDate(2019, 7, 8), QtCore.QDateTime(2019, 7, 8, 12, 0, 12)]])
-                sql = "SELECT time FROM %s" % table
+                sql = "SELECT time_t FROM %s" % table
                 res = conn.executeSql(sql)
-                self.assertIn(res, ([[QtCore.QTime(12, 0, 13)]], [['12:00:13.00']]))
-                sql = "DELETE FROM %s WHERE string = 'QGIS Rocks - \U0001f604'" % table
+
+                # This does not work in MSSQL and returns a QByteArray, we have no way to know that it is a time
+                # value and there is no way we can convert it.
```

Comment:
The provider iterator has code which does this -- but I guess the issue here is that you don't have the field type information to use to detect this situation?

Reply:
I've investigated this issue for hours, I think there is a bug upstream in QT or in the ODBC driver, because according to QT docs `time` must be supported. And yes, we don't have a field here we can guess the type from, we could check the QByteArray size and try a reinterpret_cast conversion but it's far from being a solid approach.

Reply:
Yep, leave it as is I say, it's the right decision